core/src/main/scala/algebra/ring/DivisionRing.scala

```diff
@@ -0,0 +1,22 @@
+package algebra
+package ring
+
+import scala.{specialized => sp}
+
+trait DivisionRing[@sp(Byte, Short, Int, Long, Float, Double) A] extends Any with Ring[A] with MultiplicativeGroup[A] {
+  self =>
+
+  def fromDouble(a: Double): A = Field.defaultFromDouble[A](a)(self, self)
```

Comment:
Is there any law on this? Seems odd to me a division ring only has this extra method. We don't have any other division like operation?

Reply:
Field has it too. I should have `fromDouble` in DivisionRing and make `Field` inherit `DivisionRing`.

Basically, if you have a type `A` with a function `BigInt -> A` and division, you have `fromDouble`.